tests/neg-custom-args/captures/puretest.scala

```diff
@@ -0,0 +1,10 @@
+import caps.Pure
+def foo[C <: Pure]() = ()
+def bar[T, C <: Iterable[T] & Pure]() = ()
+def baz[CC[_] <: Pure]() = ()
+def bam[CC[A] <: Pure & Iterable[A]]() = ()
```

Comment:
```suggestion
def bam[CC[A] <: Pure & Iterable[A]]() = ()
def bee[CC[A] <: Iterable[A] with Pure] = ()
```
should test this `with` as well